util/config_stripper.py

```diff
@@ -78,11 +95,61 @@ def strip_tar(input, output):
     shutil.rmtree(tempdir)
     return 0
 
-
-def strip_config(path):
+def strip_layer(path):
+    # The original layer tar is of the form <random string>/layer.tar, the
+    # working directory is one level up from where layer.tar is.
+    original_dir = os.path.normpath(os.path.join(os.path.dirname(path), '..'))
+
+    buf = cStringIO.StringIO()
+
+    # Go through each file/dir in the layer
+    # Set its mtime to 0
+    # If it's a file, add its content to the running buffer
+    # Add it to the new gzip'd tar.
+    with tarfile.open(name=path, mode='r') as it:
+      with tarfile.open(fileobj=buf, mode='w') as ot:
+        ti = it.next()
+        while ti is not None:
+          ti.mtime = 0
+          if ti.isfile():
+            f = it.extractfile(ti)
+            ot.addfile(ti, f)
+          else:
+            ot.addfile(ti)
+          ti = it.next()
+
+    # Create the new diff_id for the config
+    tar = buf.getvalue()
+    diffid = hashlib.sha256(tar).hexdigest()
+    diffid = 'sha256:%s' % diffid
+
+    # Compress buf to gz
+    gzip_process = subprocess.Popen(
+        ['gzip', '-nf'],
+        stdout=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    gz = gzip_process.communicate(input=tar)[0]
+
+    # Calculate sha of layer
+    sha = hashlib.sha256(gz).hexdigest()
+    new_name = 'sha256:%s' % sha
+    with open(os.path.join(original_dir, new_name), 'w') as out:
+      out.write(gz)
+
+    shutil.rmtree(os.path.dirname(path))
+    return (new_name, diffid)
+
+
+def strip_config(path, new_diff_ids):
     with open(path, 'r') as f:
         config = json.load(f)
     config['created'] = _TIMESTAMP
+    config['rootfs']['diff_ids'] = new_diff_ids
+
+    # Hardcode the base container name so configs will come out identical
+    config['container'] = '17aba11f7f94e6826c3144158537f0b82f8726e16dec815dc8e485f2ef1791eb'
```

Comment:
Where did these come from?

Reply:
Is there an assumption base image id is fixed?
What is the base image id which corresponds to this config sha?

Can we read this from somewhere?

Reply:
This value is essentially incorrect right now. This is an old sha of an old copy of vanilla ubuntu. I'm not sure how to make this correct but also reproducible.

I think it would involve having a constant vanilla ubuntu/debian sha to pull from and picking the correct one in the code.

Reply:
Dumb question - what if you just delete this?

The OCI spec doesn't contain this field: https://github.com/opencontainers/image-spec/blob/master/config.md#image-json

Reply:
Deleting it totally works. Done.